src/frontends/src/json_util.hpp

```diff
@@ -0,0 +1,203 @@
+#include <rapidjson/document.h>
+#include <clipper/datatypes.hpp>
+#include <stdexcept>
+
+using clipper::Input;
+using clipper::InputType;
+using clipper::Output;
+using clipper::OutputType;
+using clipper::VersionedModelId;
+using rapidjson::Type;
+
+static std::vector<std::string> kTypeNames = {
+    "Null", "False", "True", "Object", "Array", "String", "Number"};
+
+// RapidJSON value types
+/*
+enum Type {
+  kNullType = 0,      // null
+  kFalseType = 1,     // false
+  kTrueType = 2,      // true
+  kObjectType = 3,    // object
+  kArrayType = 4,     // array
+  kStringType = 5,    // string
+  kNumberType = 6     // number
+}; */
+
+namespace clipper_json {
+
+class json_parse_error : public std::runtime_error {
+ public:
+  json_parse_error(const std::string& what) : std::runtime_error(what) {}
+  ~json_parse_error() throw(){};
+};
+
+class json_semantic_error : public std::runtime_error {
```

Comment:
It may be useful in the future, but just distinguishing between semantic errors and parse errors is a good start. Can we provide the expected schema in the case of a semantic error?